core/src/main/kotlin/com/tschuchort/compiletesting/JavacUtils.kt

```diff
@@ -59,48 +57,13 @@ internal class StringJavaFileObject(className: String, private val contents: Str
     override fun getLastModified(): Long = _lastModified
 }
 
-/**
- * Gets the version string of a javac executable that can be started using the
- * given [javacCommand] via [Runtime.exec].
- */
-internal fun getJavacVersionString(javacCommand: String): String {
-    val javacProc = ProcessBuilder(listOf(javacCommand, "-version"))
-        .redirectErrorStream(true)
-        .start()
-
-    val buffer = Buffer()
-
-    javacProc.inputStream.copyTo(buffer.outputStream())
-    javacProc.waitFor()
-
-    val output = buffer.readUtf8()
-
-    return parseVersionString(output) ?: throw IllegalStateException(
-        "Command '$javacCommand -version' did not print expected output. " +
-                "Output was: '$output'"
-    )
+internal fun isJavac9OrLater(): Boolean {
+    return isJavac9OrLater(System.getProperty("java.specification.version"))
```

Comment:
What is the difference between `java.version` and `java.specification.version`? Is this the version name of the system Java installation or the one that the program runs under?

Reply:
Version includes bugfixes/patch updates, but the specification is the spec they all claim to adhere to. For all intents and purposes, this only cares about the spec version (i.e.  this function is `isJavac9OrLater`, not `isJavac904OrLater`)